backend/src/routes/RestaurantRoute.ts

```diff
@@ -0,0 +1,16 @@
+import express from 'express';
+import { param } from 'express-validator';
+import RestaurantController from '../controllers/RestaurantController';
+import { jwtCheck, jwtParse } from '../middleware/auth';
+
+const router = express.Router();
+
+router.get(
+  '/search/:city',
+  param('city').isString().trim().notEmpty().withMessage('City is required'),
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/35#issuecomment-2404204519>#3751fb</a><span> - 10/10/2024, 07:03 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The current validation for the 'city' parameter is insufficient. It only checks if the input is a non-empty string, which may allow invalid city names or potentially malicious inputs.
</div>
<div id="fix">
<b>Fix:</b> Implement more robust validation for the 'city' parameter. Use a combination of regex pattern matching, length restrictions, and potentially a whitelist of valid city names if feasible.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -7,9 +7,13 @@
  const router = express.Router();
 
  router.get(
    '/search/:city',
 -  param('city').isString().trim().notEmpty().withMessage('City is required'),
 +  param('city')
 +    .isString().trim().notEmpty().withMessage('City is required')
 +    .isLength({ min: 2, max: 50 }).withMessage('City name must be between 2 and 50 characters')
 +    .matches(/^[a-zA-Z\s-]+$/).withMessage('City name can only contain letters, spaces, and hyphens')
 +    .customSanitizer(value => value.toLowerCase()),
    jwtCheck,
    jwtParse,
    RestaurantController.searchRestaurants,
  );
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged